lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py

```diff
@@ -0,0 +1,188 @@
+            self._s1ap_wrapper._s1_util.issue_cmd(
+                s1ap_types.tfwCmd.UE_IDENTITY_RESP, us_identity_resp
+            )
+
+    def handle_detach_timer(self, req):
+        print("Detaching ", req.ue_id)
+        self.send_ue_detach(req.ue_id)
+        attachTime = random.uniform(self.attach_delay_t0, self.attach_delay_t1)
+        attachTimer = threading.Timer(attachTime, self.handle_attach_timer, args=(req,))
+        attachTimer.start()
+
+    def handle_attach_timer(self, req):
+        print("Attaching ", req.ue_id)
+        attachDuration = random.uniform(self.attach_duration_t0, self.attach_duration_t1)
+        detachTimer = threading.Timer(attachDuration, self.handle_detach_timer, args=(req,))
+        detachTimer.start()
+        self.send_attach_req(req.ue_id)
+
+    def send_ue_detach(self, ueId):
+        detach_req = s1ap_types.uedetachReq_t()
+        detach_req.ue_Id = ueId
+        detach_req.ueDetType = (
+            s1ap_types.ueDetachType_t.UE_SWITCHOFF_DETACH.value
+        )
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_DETACH_REQUEST, detach_req
+        )
+
+    def ue_thread_function(self, req, queue):
+        print("Thread starting ", req.ue_id)
+        attachTime = random.uniform(self.attach_delay_t0, self.attach_delay_t1)
+        attachTimer = threading.Timer(attachTime, self.handle_attach_timer, args=(req,))
+        attachTimer.start()
+
+        while True:
+            message = queue.get()
+            self.ue_handle_msg(message)
+
+        print("Thread finishing ", req.ue_id)
+
+    @staticmethod
+    def get_ue_id(msg):
+        if msg.msg_type == s1ap_types.tfwCmd.UE_AUTH_REQ_IND.value:
+            m = msg.cast(s1ap_types.ueAuthReqInd_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_SEC_MOD_CMD_IND.value:
+            m = msg.cast(s1ap_types.ueSecModeCmdInd_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_ESM_INFORMATION_REQ.value:
+            m = msg.cast(s1ap_types.ueEsmInformationReq_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.INT_CTX_SETUP_IND.value:
+            m = msg.cast(s1ap_types.FwNbIntCtxSetupInd_t)
+            return m.ueId
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_EMM_INFORMATION.value:
+            m = msg.cast(s1ap_types.ueEmmInformation_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND.value:
+            m = msg.cast(s1ap_types.ueAttachAccept_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_CTX_REL_IND.value:
+            m = msg.cast(s1ap_types.FwNbUeCtxRelInd_t)
+            return m.ueId
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_IDENTITY_REQ_IND.value:
+            m = msg.cast(s1ap_types.ueIdentityReqInd_t)
+            return m.ue_Id
+
+        return 0
+
+    def test_continuous_random_attach(self):
```

Comment:
is the test intended to run forever? can you add a test duration?

Reply:
That was the use case for me so far. But, makes sense to add a test duration. Let me do that.